File: ts/examples/simple-example-ts/src/functions/fibonacciFunction.ts
import {FunctionParameter, Identifier, Name} from "@code0-tech/hercules";
import {FibonacciRuntimeFunction} from "./fibonacciRuntimeFunction.js";

@Identifier("fibonacci")
@Name({code: "en-US", content: "Compute Fibonacci Number"})
@FunctionParameter({
    runtimeName: "test",
    name: [{code: "en-US", content: "Input Number"}],
    description: [{code: "en-US", content: "The position in the Fibonacci sequence"}],
    defaultValue: 10,
})
export class FibonacciFunction extends FibonacciRuntimeFunction {}

File: py/examples/simple-example-py/functions/fibonacci_function.py
from hercules import Identifier, Name, Parameter

from functions.fibonacci_runtime_function import FibonacciRuntimeFunction


@Identifier("fibonacci")
@Name({"code": "en-US", "content": "Compute Fibonacci Number"})
@Parameter(
    {
        "runtime_name": "test",
        "name": [{"code": "en-US", "content": "Input Number"}],
        "description": [
            {"code": "en-US", "content": "The position in the Fibonacci sequence"}
        ],
        "default_value": 10,
    }
)
class FibonacciFunction(FibonacciRuntimeFunction):
    pass
